app/routes/applications.py

```diff
@@ -1,28 +1,89 @@
+            key = key.upper()  # transform the key to uppercase to match zeep viva json object
+            initial_data[key] = parse_application_data(
+                value, period, parent_key=key, initial_data=[]
+            )
+
+        if parent_key in cost_types and type(value) is dict:
+            # create a single income or expense and add it to the array passed thorugh initial_data.
+            applies_to = 'coapplication' if 'Partner' in key else 'applicant'
+            period_string = f'{period["start_date"]} - {period["end_date"]}'
+
+            initial_data.append(
+                {
+                    parent_key[:-1]: {
+                        'TYPE': key,
+                        'APPLIESTO': applies_to,
+                        'PERIOD': period_string,
+                        'FREQUENCY': 12,
+                        'AMOUNT': value['amount'],
+                        'DATE': value['date'],
+                        'DESCRIPTION': '',
+                    }
+                }
+            )
+
+    return initial_data
 
 
 class Applications(Resource):
     def get(self):
         return 'APPLICATIONS LIST'
 
     def post(self):
-        json_payload = parser.parse_args()
+        json_payload = request.json
+
+        application_schema = ApplicationSchema()
+        try:
+            validated_data = application_schema.load(json_payload)
+        except ValidationError as error:
+            return jsonify(error.messages)
+
+        application_data = parse_application_data(
+            data=validated_data['application_body'],
+            period=validated_data['period'],
+            initial_data={
+                'RAWDATA': '',
+                'RAWDATATYPE': 'PDF',
+                'HOUSEHOLDINFO': '',
+                'OTHER': ''
+            }
+        )
+
+        personal_number = decode_hash_personal_number(
+            hash_id=validated_data['personal_number'])
+
+        if current_app.config['ENV'] == 'development' or current_app.config['ENV'] == 'test':
+            personal_number = make_test_personal_number(personal_number)
 
         application = VivaApplication(
-            application_type=json_payload.application_type,
-            user=decode_hash_personal_number(hash_id=json_payload.user_hash),
-            application_data=json_payload.data
+            application_type=validated_data['application_type'],
+            application_data=application_data,
+            personal_number=personal_number,
+            client_ip=validated_data['client_ip'],
+            workflow_id=validated_data['workflow_id'],
+            period=validated_data['period'],
         )
 
         response = application.create()
 
-        return jsonify(response)
+        response_schema = ResponseSchema()
+        try:
+            validated_response = response_schema.load(response)
+            return validated_response
+        except ValidationError as error:
+            return jsonify(error.messages)
```

Comment:
We need better error return than the default from viva.

Reply:
Will be done in later sprint. This will do for now. :)